vispy/visuals/mesh.py

```diff
@@ -136,6 +127,21 @@
 """)
 
 
+_null_color_transform = 'vec4 pass(vec4 color) { return color; }'
+_clim = 'float cmap(float val) { return (val - $cmin) / ($cmax - $cmin); }'
+
+
+def _build_color_transform(data, cmap, clim=(0., 1.)):
+    if data.ndim == 2 and data.shape[1] == 1:
+        fun = Function(_clim)
+        fun['cmin'] = clim[0]
+        fun['cmax'] = clim[1]
+        fun = FunctionChain(None, [fun, Function(cmap.glsl_map)])
+    else:
+        fun = Function(_null_color_transform)
+    return fun
```

Comment:
These are the bits of Python code necessary to build the `$color_transform` variable above. If it's not a colormap, the `_null_*` is used, which is just a pass-through. If it is a colormap, then we chain the colormap's `glsl_map` (which is GLSL code) after applying some simlpe GLSL code to set upper and lower limits to the data (`clim`).